models/issue.go

```diff
@@ -1207,6 +1208,10 @@ func (opts *IssuesOptions) setupSession(sess *xorm.Session) {
 		sess.And("issue.is_pull=?", false)
 	}
 
+	if opts.ExcludeArchivedRepos == util.OptionalBoolTrue {
+		sess.Join("INNER", "repository", "issue.repo_id = repository.id").And(builder.Eq{"repository.is_archived": false})
+	}
```

Comment:
```suggestion
	if opts.IsArchived != util.OptionalBoolNone {
		sess.Join("INNER", "repository", "issue.repo_id = repository.id").And(builder.Eq{"repository.is_archived": opts.IsArchived.IsTrue()})
	}
```
with suggestion above ... you an explicit require or exclude optionaly